### PR TITLE
Fix #275218: Fix more MSVC warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Done so far:

* Fix some warnings C4456 (66 of 244)
* Fix some warnings C4458 (39 of 163)
* Fix the remaining 2 warnings C4702 (show only in "Debug" mode, not in "RelWithDebInfo" mode)
* Disable warning C4703 in thirdparty/poppler, which shows when building in "RelWithDebInfo" mode

To do (by someone else, or by me after vacation):

* Disable the 7 new warnings C4189 in thirdparty/portmidi, shown when building in "RelWithDebInfo" mode
* Fix the 188 remaining C4456 (2 less in "RelWithDebInfo" mode)
* Fix the 124 remaining C4458 (1 less in "RelWithDebInfo" mode)

Ignored so far (as being code in flux, I believe):

 * Warning C4100 (introduced with 411b463)

### CODE REVIEW

File: mscore/importgtp-gp4.cpp [hunk 1]
```diff
@@ -1039,10 +1039,11 @@ bool GuitarPro4::read(QFile* fp)
                                                                               createSlur(false, chord->staffIdx(), chord);
                                                                               }
                                                                         }
+                                                                  br = true;
+                                                                  break;
                                                                   }
-                                                            br = true;
-                                                            break;
```
Comment (Jojo-Schmitz) [inline mscore/importgtp-gp4.cpp]: @lasconic : please review

**Comment by Jojo-Schmitz** [comment]
You mean [lines 1095-1111](https://github.com/musescore/MuseScore/blob/44304ca2fbe173084b4b6356c2d8f080c3284ef9/mscore/importgtp-gp4.cpp#L1095-L1111) to before [the `{`on line 1094]([https://github.com/musescore/MuseScore/blob/44304ca2fbe173084b4b6356c2d8f080c3284ef9/mscore/importgtp-gp4.cpp#L1094)?

**Comment by mattmcclinch** [comment]
It appears that I have different line numbers. I did add another brace after line 1051.

**Comment by Jojo-Schmitz** [comment]
like this?

**Comment by mattmcclinch** [comment]
Are you missing a `}` after [line 1107](https://github.com/musescore/MuseScore/blob/b1bb482f880447456788071754b72299b1669390/mscore/importgtp-gp4.cpp#L1107)?

**Comment by Jojo-Schmitz** [comment]
yes :-)

**Comment by Jojo-Schmitz** [comment]
darn, still a C4702 warning

**Comment by mattmcclinch** [comment]
That was not exactly what I had in mind. You moved that chunk of code inside [this for loop](https://github.com/musescore/MuseScore/blob/80bcf26b97748d565ade9b654cfd1633aedb3da4/mscore/importgtp-gp4.cpp#L1085), which seems wrong to me. I think the idea is to not add a glissando if the note has a Scoop or Plop ChordLine attached to it. The loop must check all elements in _nt_, and if it finds a Scoop or Plop, set _br_ to true. If _br_ is true, the glissando should not be added.

**Comment by Jojo-Schmitz** [comment]
Ok, I think now I got it right

**Comment by mattmcclinch** [comment]
That looks more like it. Do you know yet if it took care of the warning?

**Comment by Jojo-Schmitz** [comment]
Yes it did. Thanks for your help here, it was a bit of a rough ride. I shouldn't try to make last minute changes before vacation ;-)

**Comment by Jojo-Schmitz** [comment]
Continued in #3913

**Pull request merged**